### ISSUE TITLE
Isolated domain for cookie tests?

### ISSUE THREAD

**Issue by mikewest** [web-platform-tests/wpt#8581]
We're looking into porting the IETF's cookie test suite (https://github.com/abarth/http-state/tree/master/tests) to WPT, which would be simplified if we could replicate its domain structure (https://github.com/abarth/http-state/blob/master/tests/data/hosts):

```
127.0.0.1 example.org
127.0.0.1 home.example.org
127.0.0.1 home.example.org.
127.0.0.1 sibling.example.org
127.0.0.1 subdomain.home.example.org
127.0.0.1 sibling.home.example.org
```

This basically boils down to two requests:

1.  Since y'all are looking into adding a new registrable domain in https://github.com/w3c/web-platform-tests/issues/2669, perhaps you could add a "just for cookie tests" domain? That would make the checks quite a bit simpler, as we wouldn't need to deal with noise from other cookies set by other tests on the same registrable domain.

2.  Can we add a subdomain of a subdomain (and another subdomain of the same subdomain) to the list of hosts?

**Comment by foolip** [comment]
@gsnedders, can you see where this fits on your list of roadmap bugs? Seems like this could be solved together with https://github.com/w3c/web-platform-tests/issues/2669?

**Comment by gsnedders** [comment]
@foolip Yes, easy to do alongside #2669.

I'd be curious as to what @jgraham thinks of adding such a domain?

**Comment by foolip** [comment]
@gsnedders, this came up in issue triage today. With https://github.com/w3c/web-platform-tests/pull/8614 now landed, I guess a fix for this is pretty simple?

Also ping @jgraham for the requested opinions.

**Comment by gsnedders** [comment]
@mikewest How do you want to enforce the "just for cookie tests" side? A lint that allows it only in `cookies/`? That said, given the tests can be run in any order, is there actually much advantage of limiting it to just `cookies/`? Ultimately, any arbitrary cookie test could've run before it and set any arbitrary cookie, so I'm not sure what the advantage of limiting it to just cookie tests is?

**Comment by mikewest** [comment]
I wasn't thinking there would be any technical constraint for non-cookie tests. It might be nice to have a non-technical constraint (like, naming it 'for-cookies' or 'this-is-mikes-domain-go-away' or 'i-promise-this-is-a-cookie-related-test' or 'cookies-only' or 'cookies-are-the-best-really' or 'tests-that-do-not-use-cookies-should-not-use-this-domain-but-could-i-guess-if-they-really-wanted-to').

For the underlying question: I know the cookie tests we're writing will clean up after themselves, exactly because ordering would otherwise be a problem. That said, they'll still be affected by other tests (for example: if you're running ~100 tests in parallel, you'll almost certainly hit browsers' eviction limits, which is unfortunate). *shrug* Suggestions welcome!

**Comment by ahmadmiz** [comment]
#10166

**Comment by jgraham** [comment]
How important is it that we get the exact list of [sub]domains above (we can't have `.org` in any case)? It seems messy to add a lot of additional subdomains if we could instead add e.g. `child1.www` and `child2.www` and otherwise just use the existing domain set.

**Comment by mikewest** [comment]
@jgraham: I agree. The specific domains listed in the existing tests are completely unimportant. As I suggested in point 2 in the initial request, adding a subdomain of a subdomain (and another subdomain of the same subdomain) to the list of hosts is enough.

**Comment by foolip** [comment]
@gsnedders, is https://github.com/web-platform-tests/wpt/pull/10953 a full fix for what remains here (https://github.com/web-platform-tests/wpt/issues/8581#issuecomment-390888241), since there's both foo.bar and a.foo.bar in there?

**Comment by gsnedders** [comment]
The question is whether we want to fix the first item above.

**Comment by foolip** [comment]
Do you mean the "just for cookie tests" domain? I take it that would be a separate domain entirely, so that we'd require 3 domains for a setup like web-platform-tests.live? not-web-platform-tests.live already exists. cc @jugglinmike

**Comment by gsnedders** [comment]
That's my understanding of @mikewest's request.

We had a discussion about this before and I think the conclusion was that it wasn't necessary, and I don't think it makes much practical difference.

**Comment by mikewest** [comment]
I don't think we need another registrable domain. At least, not for cookies.

**Comment by gsnedders** [comment]
Then closing this now #10953 has landed.

**Comment by gsnedders** [comment]
Reopening with https://github.com/web-platform-tests/wpt/pull/12462.

**Comment by jugglinmike** [comment]
Commit 4f0b8fb reverted the fix, so this issue should remain open.

**Comment by gsnedders** [comment]
FFS GitHub, closing it in the revert commit.

**Comment by foolip** [comment]
@gsnedders can you provide an update on this issue? Thanks!

**Comment by gsnedders** [comment]
We landed extra subdomains, reverted them, and we now have #13272 open to reland.
